Sources/JavaScriptKit/FundamentalObjects/JSFunction.swift

```diff
@@ -93,15 +93,16 @@ internal func invokeJSFunction(_ jsFunc: JSFunction, arguments: [ConvertibleToJS
             var payload1 = JavaScriptPayload1()
             var payload2 = JavaScriptPayload2()
             if let thisId = this?.id {
-                _call_function_with_this(thisId,
+                _call_function_with_this_unsafe(thisId,
                                          jsFunc.id, argv, Int32(argc),
                                          &kindAndFlags, &payload1, &payload2)
             } else {
-                _call_function(
+                _call_function_unsafe(
                     jsFunc.id, argv, Int32(argc),
                     &kindAndFlags, &payload1, &payload2
                 )
             }
+            assert(!kindAndFlags.isException)
             let result = RawJSValue(kind: kindAndFlags.kind, payload1: payload1, payload2: payload2)
             return (result.jsValue(), kindAndFlags.isException)
         }
```

Comment:
one small change here — you don’t need to handle the exception case once you’ve `assert()`-ed it away:

```swift
private func invokeNonThrowingJSFunction(...) throws -> JSValue {
    arguments.withRawJSValues { rawValues in
        rawValues.withUnsafeBufferPointer { bufferPointer -> JSValue in
            // ...
            assert(!kindAndFlags.isException)
            let result = RawJSValue(kind: kindAndFlags.kind, payload1: payload1, payload2: payload2)
            return result.jsValue()
        }
    }
}
```

Reply:
Also `throws` is not required :)

Reply:
Yeap you are absolutely right! I've applied your suggestions in a new commit 😃